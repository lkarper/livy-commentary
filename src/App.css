.App {
    width: 100%;
}

main {
    padding: 0;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-bottom: 50px;
}

.section {
    width: 100%;
    padding: 10px;
    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;
}