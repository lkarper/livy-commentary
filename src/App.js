import React from 'react';
import { Route } from 'react-router-dom';
import Header from './Header/Header';
import CommentaryHome from './CommentaryHome/CommentaryHome';
import CommentaryRead from './CommentaryRead/CommentaryRead';
import './App.css';

const App = (props) => {
  return (
    <div className="App">
      <Header />
      <main>
        <Route 
          exact path="/"
          component={CommentaryHome}
        />
        <Route 
          path='/commentary-read/:location'
          component={CommentaryRead}
        />
      </main>
    </div>
  );
}

export default App;
