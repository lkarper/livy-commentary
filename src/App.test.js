import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from './App';

describe('App component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <App />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <App />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
