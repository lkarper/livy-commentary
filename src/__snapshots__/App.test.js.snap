// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <div
        className="App"
      >
        <Header
          forceUpdate={[Function]}
        >
          <header
            className="Header__header"
          >
            <h1
              className="Header__h1"
            >
              <Link
                className="Header__h1-a"
                to="/"
              >
                <LinkAnchor
                  className="Header__h1-a"
                  href="/"
                  navigate={[Function]}
                >
                  <a
                    className="Header__h1-a"
                    href="/"
                    onClick={[Function]}
                  >
                    The Third Decade Project
                  </a>
                </LinkAnchor>
              </Link>
            </h1>
          </header>
        </Header>
        <main>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          >
            <CommentaryHome
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div
                className="CommentaryHome__book-link-container"
              >
                <p
                  className="CommentaryHome__loading p"
                >
                  Loading...
                </p>
              </div>
            </CommentaryHome>
          </Route>
          <Route
            component={[Function]}
            path="/commentary-read/:location"
          />
          <PublicOnlyRoute
            path="/admin-login"
            render={[Function]}
          >
            <Route
              path="/admin-login"
              render={[Function]}
            />
          </PublicOnlyRoute>
          <PrivateRoute
            component={[Function]}
            path="/edit"
          >
            <Route
              path="/edit"
              render={[Function]}
            />
          </PrivateRoute>
        </main>
      </div>
    </App>
  </Router>
</BrowserRouter>
`;
